### PR TITLE
Refactored to rename "service terms" to "terms of service"

### PR DESCRIPTION
**Pull request by harshilsharma**
#### Summary
Refactored to rename "service terms" to "terms of service"

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-12400

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-server/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by harshilsharma** [comment]
@lieut-data I know one `User.LatestAcceptedServiceTerms`, which is being removed as part of another issue along with this. What's the other?

File: api4/user.go [hunk 1]
```diff
@@ -1622,23 +1622,24 @@ func enableUserAccessToken(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
-func registerServiceTermsAction(c *Context, w http.ResponseWriter, r *http.Request) {
+func registerTermsOfServiceAction(c *Context, w http.ResponseWriter, r *http.Request) {
 	props := model.StringInterfaceFromJson(r.Body)
 
 	userId := c.Session.UserId
-	serviceTermsId := props["serviceTermsId"].(string)
+	// TODO refactor this to be terms of service
```
Comment (lieut-data) [inline api4/user.go]: @harshilsharma: TODO here re: discussion.

File: app/config.go [hunk 2]
```diff
@@ -246,12 +246,13 @@ func (a *App) AsymmetricSigningKey() *ecdsa.PrivateKey {
...
-	if a.clientConfig["EnableCustomServiceTerms"] == "true" {
-		serviceTerms, err := a.GetLatestServiceTerms()
+	if a.clientConfig["EnableCustomTermsOfService"] == "true" {
+		termsOfService, err := a.GetLatestTermsOfService()
 		if err != nil {
 			mlog.Err(err)
 		} else {
-			a.clientConfig["CustomServiceTermsId"] = serviceTerms.Id
+			// TODO refactor this to be terms of service
+			a.clientConfig["CustomTermsOfServiceId"] = termsOfService.Id
```
Comment (lieut-data) [inline app/config.go]: @harshilsharma: TODO here re: discussion.

File: model/terms_of_service.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	"fmt"
+	"io"
+	"net/http"
+	"unicode/utf8"
+)
+
+// we only ever need the latest version of terms of service
+const TERMS_OF_SERVICE_CACHE_SIZE = 1
+
+// TODO refactor this to be terms of service
```
Comment (lieut-data) [inline model/terms_of_service.go]: @harshilsharma: TODO here, but seems spurious now?

File: api4/user.go (hunk 1 above)
Comment (harshilsharma) [reply api4/user.go]: Ah yes, the TODOs have already been implemented. Removing them.

File: app/config.go (hunk 2 above)
Comment (harshilsharma) [reply app/config.go]: Removed as already done.

File: model/terms_of_service.go (hunk 3 above)
Comment (harshilsharma) [reply model/terms_of_service.go]: Removing as already done.

**Comment by harshilsharma** [comment]
@lieut-data those TODOs were already done but the TODO comment wasn't removed. SO just one TODO left, about the `User.AcceptedServiceTermsId` as discussed above.

**Comment by lieut-data** [comment]
I've pushed some tweaks (including a bug fix) that completes the rename, even for the table name and columns since we can patch pre-release and all the RC instances get re-created anyway.

**Pull request merged**